overlord/snapstate/snapstate.go

```diff
@@ -300,6 +300,23 @@ func Info(s *state.State, name string, revision int) (*snap.Info, error) {
 	return nil, fmt.Errorf("cannot find snap %q at revision %d", name, revision)
 }
 
+// Current returns the information about the current revision of a snap with the given name.
+func Current(s *state.State, name string) (*snap.Info, error) {
```

Comment:
this can be simplified a bit:

if err != nil && err != state.ErrNoState {
  return nil, err
}
if snapst.Current() == nil {
   return nil, fmt.Errorf("cannot find snap %q", name)
}
...